src/index.ts

```diff
@@ -771,10 +771,18 @@ export class Squid {
   }): Promise<TokenBalance[]> {
     // remove invalid and duplicate chains and convert to number
     const filteredChains = new Set(chains.map(Number).filter(c => !isNaN(c)));
+    const chainRpcUrls = this.chains.reduce(
+      (acc, chain) => ({
+        ...acc,
+        [chain.chainId]: chain.rpc
+      }),
+      {}
+    );
 
     return getAllEvmTokensBalance(
       this.tokens.filter(t => filteredChains.has(Number(t.chainId))),
```

Comment:
this is not necessary as all tokens need to have a chain id

Reply:
It is filtering tokens for only passing to `getAllEvmTokensBalance()` the tokens of the chains included in the specified `chains` array.

Reply:
For example, if provided `chains` array is `[1, 43114]`, the tokens passed to `getAllEvmTokensBalance()` are only the tokens with those chain id.